src/NLog/Targets/EventLogTarget.cs

```diff
@@ -98,6 +98,12 @@ public EventLogTarget(string name)
             Name = name;
         }
 
+        /// <summary>
+        /// Gets or sets the provider of Windows event log static methods to be utilized by <see cref="EventLogTarget"/>.
+        /// </summary>
+        /// <remarks>Introduced to improve the testability of the class.</remarks>
+        protected internal IEventLogStaticWrapper EventLogStaticMethods { get; set; } = EventLogStaticWrapper.Instance;
```

Comment:
Think it would be nicer if you implemented an internal constructor that took an internal IEventLogWrapper-interface as input parameter. The default constructor would then call the new internal constructor with a private-class-implementation of the IEventLogWrapper-interface. Ofcourse the unit-tests will not be able to test initialization from NLog.config, but I think that is okay.

Reply:
I get your point on injecting the implementation in the constructor instead of the property. But you mentioned an `IEventLogWrapper`, do you suggest merging the `IEventLogStaticWrapper` and `IEventLogInstanceWrapper`?

Also, you think these new interfaces and implementations should not fall under `protected internal`?

Reply:
> do you suggest merging the IEventLogStaticWrapper and IEventLogInstanceWrapper?

Yes I would prefer just a single interface. And this interface and the constructor getting it as parameter are both marked as `internal`-only (Not protected).